src/TeamContributionCalendar/TeamContributionCalendar.js

```diff
@@ -140,10 +143,18 @@ export default class TeamContributionCalendar {
       if (data.error) {
         console.error(data.errorMessage);
       } else {
-        this.processGitHubCalendar(data.parsedCalendar, {
-          start: user.from,
-          end: user.to
-        });
+        const timeframe = { start: user.from, end: user.to };
+        const sanitized = gitHubUtils.sanitize(data.parsedCalendar);
+
+        const filtered = sanitized.map(week =>
+          week.map(day =>
+            withinTimeframe(day.attributes["data-date"], timeframe)
+              ? day
+              : undefined
+          )
+        );
```

Comment:
It's a draft, tests fail, but let's discuss the changes first. My idea goes as follows:
- the structure of the calendars should be simplified somehow(see `sanitized()`, this basically removes the "noise" from the svg, e.g. attributes, month and weekday texts around the calendar etc)
- the calendars should be narrowed down according to the timeframe right before passing them to the `process*` functions
- `undefined` should be returned in the case of GitHub calendars to update the daily contributions at the correct position